app/models/course.rb

```diff
@@ -113,6 +113,23 @@ def root_folder?
     end
   end
 
+  def self_directed_learning_max_time
+    settings(:course).self_directed_learning_max_time
+  end
+
+  def self_directed_learning_max_time=(time)
```

Comment:
Not sure, but something other than `self_directed_learning` would be good; it's too long. 

If we wanted to be more succinct, maybe `earlier_start_at` (or something like `earlier_start_interval`), so the code would be like this:

```ruby
if course && course.earlier_start_at
  start_at && start_at - course.earlier_start_at < Time.zone.now
else
```

Maybe can ask @fonglh also? 😅